app/views/assessment/AssessmentQuestion.js

```diff
@@ -0,0 +1,142 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { SafeAreaView, ScrollView, StyleSheet, Text, View } from 'react-native';
+
+import Colors from '../../constants/colors';
+import Fonts from '../../constants/fonts';
+import AssessmentButton from './AssessmentButton';
+import { AnswersContext } from './AssessmentContext';
+import AssessmentOption from './AssessmentOption';
+import {
+  QUESTION_TYPE_MULTI,
+  SCREEN_TYPE_CHECKBOX,
+  SCREEN_TYPE_DATE,
+  SCREEN_TYPE_RADIO,
+} from './constants';
+
+/**
+ * @typedef { import("./Assessment").SurveyQuestion } SurveyQuestion
+ * @typedef { import("./Assessment").SurveyOption } SurveyOption
+ * @typedef { import("./Assessment").SurveyAnswers } SurveyAnswers
+ */
+
+/** @type {React.FunctionComponent<{
+ *   onNext: () => void;
+ *   onChange: (value: { index: number, value: string }[]) => void;
+ *   option: SurveyOption
+ *   question: SurveyQuestion
+ * }>} */
+const AssessmentQuestion = ({ onNext, onChange, option, question }) => {
+  const { t } = useTranslation();
+  const answers = useContext(AnswersContext);
+  const [selectedValues, setSelectedValues] = useState(
+    answers[question.id] || [],
+  );
+  /** @type {(value: string, index: number) => void} */
+  const onSelectHandler = (value, index) => {
+    switch (question.question_type) {
+      case QUESTION_TYPE_MULTI:
+        setSelectedValues(values => {
+          let exists = values.some(v => v.index === index);
```

Comment:
What are verifying exists here? It's hard to understand with this variable name.

Also, please use a `const` here since the value is never modified.